spring-integration-file/src/main/java/org/springframework/integration/file/remote/handler/FileTransferringMessageHandler.java

```diff
@@ -139,7 +167,22 @@ protected void onInit() throws Exception {
 
 	@Override
 	protected void handleMessageInternal(Message<?> message) throws Exception {
-		this.remoteFileTemplate.send(message, this.mode);
+		String path = this.remoteFileTemplate.send(message, this.mode);
+		if (this.chmod != null && isChmodCapable()) {
+			doChmod(this.remoteFileTemplate, path, this.chmod);
```

Comment:
Hmmm - I suppose we could add an overloaded send on the template that takes the chmod mask.

I don't really want to get involved with TLs.